src/main/java/com/amazon/opendistroforelasticsearch/sql/query/ESActionFactory.java

```diff
@@ -71,21 +74,22 @@ public static QueryAction create(Client client, String sql) throws SqlParseExcep
                 if(isMulti(sqlExpr)){
                     sqlExpr.accept(new TermFieldRewriter(client, TermRewriterFilter.MULTI_QUERY));
                     MultiQuerySelect multiSelect = new SqlParser().parseMultiSelect((SQLUnionQuery) sqlExpr.getSubQuery().getQuery());
-                    handleSubQueries(client,multiSelect.getFirstSelect());
-                    handleSubQueries(client,multiSelect.getSecondSelect());
                     return new MultiQueryAction(client, multiSelect);
                 }
+                else if(isSubquery(sqlExpr)){
+                    SubqueryRewriterHelper.rewrite(sqlExpr);
```

Comment:
This is minor, but why we do not define new rewriter and do not write the statement as a 

    sqlExpr.accept(new SubqueryRewriter())? (for example it's done on the top for NesttedFieldRewriter)?

for more consistent codebase?

this is minor, I'm just curious

Reply:
Good question. This is actually my first choice. the major concern is this solution change the AST before the parser know whether it can support it or not. 
There are two reasons why I come with the proposed solution
- The soluiton define two steps. (1) Is it supported subquery (2) if yes, rewrite the subquery. which clearly define the query is supported or not and this structure make it easy to add more subquery use case in future
- In my understanding, we should not change the original AST, but create and optimize the logical plan for different query type. for example, SELECT * FROM A WHERE a IN (SELECT b FROM B), The AST should represent the subquery, but the logical plan should be JOIN. This soluiton make it on the right way.